api/v1alpha1/trafficsplit_types.go

```diff
@@ -45,9 +39,139 @@ type TrafficSplit struct {
 	Status TrafficSplitStatus `json:"status,omitempty" protobuf:"bytes,4,opt,name=status"`
 }
 
+// TrafficSplitSpec defines the desired state of TrafficSplit.
+type TrafficSplitSpec struct {
+	// Rules defines a schema for splitting traffic to backend
+	// referenced objects.
+	//
+	// Support: Core
+	//
+	// +required
+	Rules []TrafficSplitRule `json:"rules"`
+}
+
+// TrafficSplitRule defines the destination and rules for splitting
+// traffic.
+type TrafficSplitRule struct {
+	// Destination is a reference to an object that should be used
+	// to forward traffic.
+	//
+	// Support: Core
+	//
+	// +required
+	Destination DestinationObjectReference `json:"destination"`
+	// Weight specifies the proportion of traffic to be forwarded to a
+	// destination. A valid weight value is 0-100. The sum of weights
+	// across destinations must equal 100. If only one trafficsplit
+	// destination is specified, the default weight is 100.
```

Comment:
IMHO we should make this explicit. If Weight is nil, the TrafficSplit object should be considered invalid. 

We should also mention that is sum of weights across destination is not 100, then the TrafficSplit is considered invalid. WDYT, Daneyon?

Reply:
The behaviour I've seen for other things in Kube that take weights is that the values are totalled, then the fractions are used. I think that's what I would expect for a `weight` field.

So, if there are three weights, of 40, 40 and 40, then each service will get a third (40/120).